core/src/main/java/org/elasticsearch/action/ActionListenerResponseHandler.java

```diff
@@ -36,6 +38,20 @@ public ActionListenerResponseHandler(ActionListener<Response> listener) {
         this.listener = listener;
     }
 
+    /**
+     * Convenience method for delegating a response provided by supplier to the provided transport channel
+     */
+    public static <T extends TransportResponse> ActionListenerResponseHandler responseHandler(ActionListener<T> listener, Supplier<T>
```

Comment:
My point here is that we don't need this static method. We can make `ActionListenerResponseHandler` a proper class that takes a response supplier as argument as every usage site of this class is of the form 

```new ActionListenerResponseHandler<DfsSearchResult>(listener) {
            @Override
            public DfsSearchResult newInstance() {
                return new DfsSearchResult();
            }
        });
```